src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -508,179 +544,185 @@ public async Task<Repository> CreateECRRepository(string repositoryName)
+
+                return optionSetting;
             });
+        }
 
-            foreach (var settingDescription in response.ConfigurationSettings)
+        public async Task<Repository> DescribeECRRepository(string respositoryName)
+        {
+            var client = _awsClientFactory.GetAWSClient<IAmazonECR>();
+            return await HandleException(async () =>
             {
-                foreach (var setting in settingDescription.OptionSettings)
+
+                DescribeRepositoriesResponse response;
+                try
                 {
-                    optionSetting.Add(setting);
+                    response = await client.DescribeRepositoriesAsync(new DescribeRepositoriesRequest
+                    {
+                        RepositoryNames = new List<string> { respositoryName }
+                    });
+                }
+                catch (RepositoryNotFoundException ex)
+                {
+                    throw new AWSResourceNotFoundException(DeployToolErrorCode.ECRRepositoryDoesNotExist, $"The ECR repository {respositoryName} does not exist.", ex);
                 }
-            }
 
-            return optionSetting;
+                return response.Repositories.First();
+            });
         }
 
-        public async Task<Repository> DescribeECRRepository(string respositoryName)
+        private async Task<T> HandleException<T>(Func<Task<T>> action)
         {
-            var client = _awsClientFactory.GetAWSClient<IAmazonECR>();
-
-            DescribeRepositoriesResponse response;
             try
             {
-                response = await client.DescribeRepositoriesAsync(new DescribeRepositoriesRequest
-                {
-                    RepositoryNames = new List<string> { respositoryName }
-                });
+                return await action();
             }
-            catch (RepositoryNotFoundException ex)
+            catch (AmazonServiceException e)
             {
-                throw new AWSResourceNotFoundException(DeployToolErrorCode.ECRRepositoryDoesNotExist, $"The ECR repository {respositoryName} does not exist.", ex);
-            }
+                var messageBuilder = new StringBuilder();
+                if (!string.IsNullOrEmpty(e.ErrorCode))
+                {
+                    messageBuilder.AppendLine($"{e.ErrorCode}");
+                }
+
+                if (!string.IsNullOrEmpty(e.Message))
+                {
+                    messageBuilder.AppendLine(e.Message);
+                }
 
-            return response.Repositories.First();
+                if (messageBuilder.Length == 0)
+                {
+                    messageBuilder.Append($"An unknown error occurred while communicating with AWS.{Environment.NewLine}" +
+                                          "Re-run the command with the --diagnostics option to see the full error message.");
```

Comment:
Since users coming in from the VS Toolkit don't have an option to set `--diagnostics` I think we should remove that comment here.

Reply:
that makes sense, followup to that, is there any way toolkit customer can enable diagnostics? There should be some way, env var or something.

Reply:
We have a story in both our backlog and IDE backlog to allow the IDE pass in additional command line arguments when starting server mode.